### ISSUE TITLE
How to generate new version of elk-full.jar

### ISSUE THREAD

**Issue by fchastanet** [plantuml/plantuml#1813]
**What is the URL of the documentation?**

[elk](https://plantuml.com/elk)

**What can be improved?**

A link is provided to download elk-full.jar pointing to http://beta.plantuml.net/elk-full.jar
but this version is quite outdated 2021-04-27
Could you provide either an updated jar file, either the recipe to generate a new one ?

**Additional context**

N/A

**Comment by fchastanet** [comment]
I found a temporary solution by forking the repository https://github.com/plantuml/ELK-demo
Here my PR https://github.com/plantuml/ELK-demo/pull/5
I released a new version https://github.com/fchastanet/ELK-demo/releases/tag/v0.9.1
Here the [elk-full-0.9.1.jar](https://github.com/fchastanet/ELK-demo/releases/download/v0.9.1/elk-full-0.9.1.jar)

**Comment by arnaudroques** [comment]
Many thanks for your help!
We should probably use CI/CD and gradle on https://github.com/plantuml/ELK-demo to automatically build this elk-full-0.9.1.jar file.
If you are confortable with those, you can go on and propose an PR :-)

Regards,
Arnaud

**Comment by arnaudroques** [comment]
> Hello, I read that the `elk-full.jar` file is included in the main `plantuml.jar` in one of the newer releases (can unfortunately not find the changelog entry anymore). So, is it still required to generate this separatly?

Yes, starting from V1.2024.6, ELK is included, so no, you do not need  `elk-full.jar`  anymore.
We still have to update the documentation!

**Comment by vlsav** [comment]
Hello! elk-full.jar included into plantuml distribution seems like compiled with newer Java Runtime version.
I has no issues with drawing plantuml diagrams, but when I am trying to use ELK I got this: 
java.lang.UnsupportedClassVersionError: org/eclipse/elk/graph/util/ElkGraphUtil has been compiled by a more recent version of the Java Runtime (class file version 55.0), this version of the Java Runtime only recognizes class file versions up to 52.0

**Comment by jzabroski** [comment]
The root cause is a JVM byte code compatibility between the jar file and the Java runtime loading the jar.
```text
class file version 52.0 → Java 8
class file version 55.0 → Java 11
```
Right now, the plantuml.com editor has this error with ELK layouts:

java.lang.UnsupportedClassVersionError: org/eclipse/elk/graph/util/ElkGraphUtil has been compiled by a more recent version of the Java Runtime (class file version 61.0), this version of the Java Runtime only recognizes class file versions up to 55.0
```text
class file version 61.0 → Java 17
class file version 55.0 → Java 11
```
It would be nice to have this fixed as ELK seems to come the closest to following Richard Barker's "Crow's Foot Rule" when drawing ER Diagrams. While smetana avoids overlapping
